adapters/inmobi/inmobi.go

```diff
@@ -0,0 +1,97 @@
+package inmobi
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/mxmCherry/openrtb"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+	"net/http"
+)
+
+type InmobiAdapter struct {
+	endPoint string
+}
+
+func NewInmobiAdapter(endpoint string) *InmobiAdapter {
+	return &InmobiAdapter{
+		endPoint: endpoint,
+	}
+}
+
+func (a *InmobiAdapter) MakeRequests(request *openrtb.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	var errs []error
+
+	for _, imp := range request.Imp {
+		if imp.Banner != nil {
+			banner := imp.Banner
+			if (banner.W == nil || banner.H == nil || *banner.W == 0 || *banner.H == 0) && len(banner.Format) > 0 {
+				format := banner.Format[0]
+				banner.W = &format.W
+				banner.H = &format.H
+			}
+		}
+	}
+
+	reqJson, err := json.Marshal(request)
+	if err != nil {
+		errs = append(errs, err)
+		return nil, errs
+	}
+
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+
+	return []*adapters.RequestData{{
+		Method:  "POST",
+		Uri:     a.endPoint,
+		Body:    reqJson,
+		Headers: headers,
+	}}, errs
+}
+
+func (a *InmobiAdapter) MakeBids(interalRequest *openrtb.BidRequest, externalRequest *adapters.RequestData, response *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	if response.StatusCode == http.StatusNoContent {
+		return nil, nil
+	}
+
+	if response.StatusCode != http.StatusOK {
+		return nil, []error{&errortypes.BadServerResponse{
+			Message: fmt.Sprintf("Unexpected http status code: %d", response.StatusCode),
+		}}
+	}
+
+	var serverBidResponse openrtb.BidResponse
+	if err := json.Unmarshal(response.Body, &serverBidResponse); err != nil {
+		return nil, []error{err}
+	}
+
+	bidResponse := adapters.NewBidderResponseWithBidsCapacity(1)
+
+	for _, sb := range serverBidResponse.SeatBid {
+		for i := range sb.Bid {
+			mediaType := getMediaTypeForImp(sb.Bid[i].ImpID, interalRequest.Imp)
```

Comment:
I don't see this being a problem, but I want you to be aware that the `interalRequest` here is the same one you're editing in `MakeRequests` to set the banner W/H.

Reply:
From my understanding it shouldn't. Just following a pattern I saw in other adapters that I presume also have issue's explicitly declaring what the media type is post-bid.